src/main/java/com/fasterxml/jackson/databind/util/ClassUtil.java

```diff
@@ -1179,7 +1179,7 @@ public static boolean isJDKClass(Class<?> rawType) {
     public static boolean isJDK17OrAbove() {
         try {
             return getJDKMajorVersion() >= 17;
-        } catch (Throwable t) {
+        } catch (Exception t) {
```

Comment:
No real tests to see that this wont start failing; let's leave catching `Throwable`

Reply:
in cases where we don't want to make too big a change, could we consider something like the ExceptionUtil I documented in https://medium.com/system-weakness/cwe-248-uncaught-exception-in-java-b6f4ee8daf14 ? Errors like OutOfMemoryError and ThreadDeathError should be rethrown. LinkageError is something that we would need to consider on a case by case basis.

Reply:
Hmmh. Perhaps for cases where we can determine limits and so on, but allow-listing requires quite a bit of work to get to suitable set.

I think the idea is valid, but I feel not for this particular use case; given that we already catch safely cases during initialization: if we were starting from scratch, I would agree in doing things incrementally. But in this particular case let's keep things simple even if not optimal: I don't want accidental breakages on legacy platforms when there isn't specific problem to fix (as far as I can see).
This is assuming this is more like pro-active maintenance, general idea of trying to avoid catching `Throwable` (which I agree with, at general level )